ibm/service/satellite/resource_ibm_satellite_location.go

```diff
@@ -98,6 +98,11 @@ func ResourceIBMSatelliteLocation() *schema.Resource {
 				Optional:    true,
 				Description: "A description of the new Satellite location",
 			},
+			"coreos_enabled": {
```

Comment:
if we don't send coreos_enabled by default will it be false? If so can we mark argument as both optinal and computed
Can we set the argument in Read method ...?

Reply:
Can we add this argument to datasource also as computed

Reply:
Correct. Default is false.

I have made the changes you requested. Thanks for the review.

Out of curiosity, what does `resourceIBMSatelliteLocationRead` do? It pulls in info on the location - but where is that function used in the provider?

Reply:
resourceIBMSatelliteLocationRead is the function which will be used before any action like apply/update.. It first makes a refresh call(resourceIBMSatelliteLocationRead) and checks whether infrastructure already exists by comparing to statefile(if present)
It writes the data to statfile which helps to maintian the current state of templates..
It is used during import scenario also where he wants to bring the infrastructure which was provisioned outside Terraform and monitor using Terraform in future.
Its a key method which helps to find if any drift exists between the exisiting infrastructure and the data stored in statefile

Reply:
Thank you for the detailed explanation. That's quite helpful.